tools/pre-swap-checker/contracts/PreSwapChecker.sol

```diff
@@ -36,87 +36,93 @@ contract PreSwapChecker {
     wethContract = IWETH(preSwapCheckerWethContract);
   }
 
+
   /**
-    * @notice If order is going through delegate via provideOrder
-    * ensure necessary checks are set
+    * @notice If order is going through wrapper to a delegate
     * @param order Types.Order
     * @param delegate IDelegate
+    * @param wrapper address
     * @return uint256 errorCount if any
     * @return bytes32[] memory array of error messages
     */
-  function checkSwapDelegate(
+  function checkWrappedDelegate(
     Types.Order calldata order,
-    IDelegate delegate
+    IDelegate delegate,
+    address wrapper
     ) external view returns (uint256, bytes32[] memory ) {
-
-    bytes32[] memory errors = new bytes32[](20);
+    bytes32[] memory errors = new bytes32[](22);
     uint256 errorCount;
     address swap = order.signature.validator;
-    IDelegate.Rule memory rule = delegate.rules(order.sender.token,order.signer.token);
-    (uint256 swapErrorCount, bytes32[] memory swapErrors) = checkSwapSwap(order, false);
-
+    (uint256 swapErrorCount, bytes32[] memory swapErrors) = coreSwapChecks(order);
+    (uint256 delegateErrorCount, bytes32[] memory delegateErrors) = coreDelegateChecks(order, delegate);
     if (swapErrorCount > 0) {
       errorCount = swapErrorCount;
-      // copies over errors from checkSwapSwap to be outputted
+      // copies over errors from coreSwapChecks
```

Comment:
Do these need to be copied across? If these are the first errors found, the function could just continue adding to the array returned by `coreSwapChecks`

Reply:
good point, can take a look and refactor as needed.

Reply:
It's been refactored to reduce copying over arrays.